### PR TITLE
Add OAuth2 Authentication Support. 

### PR DESCRIPTION
**Pull request by mayur-solace**
- Update the Spring Boot dependency to 3.3.1
- Update the Solace JCSMP dependency to 10.22.0
- Added Support for enabling OAuth2 authentication scheme with the Soalce PubSub+ Broker
- Added ITs  for OAuth2 and  Client Certificate authentication 
- Updated the failsafe and surefire plugin version and updated existing  unit test to JUnit5
- Added Sample for OAuth2  authentication
- Update the developer readme to include OAuth2 support
- Updated the version number in pom.xml file

### CODE REVIEW

**Comment by mayur-solace** [comment]
@Nephery for review of the design and any suggestions from you?

File: solace-spring-boot-autoconfigure/solace-java-spring-boot-autoconfigure/pom.xml [hunk 1]
```diff
@@ -34,6 +34,12 @@
 			<version>${solace.jcsmp.version}</version>
 		</dependency>
 
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-oauth2-client</artifactId>
+		</dependency>
+
+
```
Comment (Nephery) [inline solace-spring-boot-autoconfigure/solace-java-spring-boot-autoconfigure/pom.xml]: Correct me if I'm wrong, but isn't OAuth2 usually an opt-in dependency in Spring? Maybe this should be optional here as well for users who aren't using oauth?

Comment (mayur-solace) [reply solace-spring-boot-autoconfigure/solace-java-spring-boot-autoconfigure/pom.xml]: yes this optional dependency. I'm not 100% sure how optional actually works. I will read more on optional and will make it optional.

**Comment by mayur-solace** [comment]
Will raise PR directly against https://github.com/SolaceProducts/solace-spring-boot

**Pull request closed without merging**